### PR TITLE
Lwpmlse 329 hipblas test yaml

### PR DESCRIPTION
**Pull request by TorreZuk**
* support yaml driven testing alongside original testing framework, using snake case tokens for now
* adds --yaml hipblas_smoke.yaml
* fix general testing support found using new smoke test set
* unifies style of test framework with model and parameter to stack local var pattern
* lwpmlse-329

### CODE REVIEW

File: clients/include/hipblas_test.hpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    virtual ~hipblas_test_invalid() = default;
+};
+
+/* ============================================================================================ */
+/*! \brief  Normalized test name to conform to Google Tests */
+// The template parameter is only used to generate multiple instantiations with distinct static local variables
+template <typename>
+class RocBLAS_TestName
```
Comment (daineAMD) [inline clients/include/hipblas_test.hpp:97]: A couple instances of rocblas remaining instead of hipblas.

**Review by daineAMD — COMMENTED** [review]
About halfway done reviewing, looks good so far - will finish review tomorrow.

File: clients/include/hipblas_test.hpp (hunk 1 above)
Comment (TorreZuk) [reply clients/include/hipblas_test.hpp:97]: Thanks that may be dead code, will fix.  Sorry for the PR size it just kept growing 🤣

File: clients/include/testing_proto.hpp [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+// Copyright (C) 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+// Script-generated file -- do not edit
+
+#pragma once
+
+#include "hipblas_arguments.hpp"
+
+template <typename T>
+inline inline hipblasStatus_t testing_asum(const Arguments& arg);
```
Comment (daineAMD) [inline clients/include/testing_proto.hpp:11]: are these supposed to be `inline inline`?

Comment (TorreZuk) [reply clients/include/testing_proto.hpp:11]: I think it should be "really inline"  :-) Very good find, more dead code.  That design didn't work as the template prototypes usage didn't work as compiler needed the implementation for the dispatch function pointers, will delete the file.

**Pull request merged**